content/advent-2016/saga-go-dependency-management.md

```diff
@@ -0,0 +1,106 @@
+
+Mandate notwithstanding, it wasn’t clear to leaders in the community how we’d take the first step. So, Peter Bourgon stepped up to break the collective paralysis: he offered to convene a committee that would work out a proposal for an official, unified tool. That became [a committee plus an advisory group](https://docs.google.com/document/d/18tNd8r5DV0yluCR7tPvkMTsWD_lYcRO7NhpNSDymRr8). Between the two, perspectives and experience from all the major tools have been on hand.
+
+## The package management committee
+
+If you've spent much time in the world of software, "a committee to develop a unified tool" probably sets your mental alarm bells a-jangling. It certainly did for those of us on the committee, and we've tried to manage that risk from our very first meeting.
+
+One step we've taken is to publicly release our work as soon as we felt it could engender productive discussion. That's worked out quite well. The comments we've received have been invaluable, and led us to shift direction more than once. We've also relied heavily on excellent research and feedback from the advisory group.
+
+Here's a quick history of the committee's work from our first meeting in early September:
+
+* To get the conversation started, we each created some imaginary CLI backscroll describing how a user would work with our ideal tool
+* We then laid out a set of [major user stories](https://docs.google.com/document/d/1wT8e8wBHMrSRHY4UF_60GCgyWGqvYye4THvaDARPySs)
+* Working from the user stories, we explored the [design space](https://docs.google.com/document/d/1TpQlQYovCoX9FkpgsoxzdvZplghudHAiQOame30A-v8) of the problem
+* From the user stories and design space, we created a (working) [specification](https://docs.google.com/document/d/1qnmjwfMmvSCDaY4jxPmLAccaaUI5FfySNE90gB0pTKQ/edit#heading=h.4d61hnb1y8gc) for a tool
+* We're now iterating on tool implementation, with [gps](https://github.com/sdboyer/gps) as the engine. We’re learning and tweaking as we go, and plan to open up the repository publicly in early January
+
+The [original plan](https://docs.google.com/document/d/18tNd8r5DV0yluCR7tPvkMTsWD_lYcRO7NhpNSDymRr8) was for the committee to deliver a more formal proposal, then begin implementation. However, it became clear over the course of our discussions that a proposal absent implementation would result in exactly the kind of design-by-committee waterfall that we wanted to avoid. So, while there is a proposed spec, please expect it to evolve as we implement and learn.
+
+### One tool, to unite them all
+
+Our goal for the first iteration is a tool that’s minimal, but covers the fundamental requirements. The [spec doc](https://docs.google.com/document/d/1qnmjwfMmvSCDaY4jxPmLAccaaUI5FfySNE90gB0pTKQ/edit#heading=h.4d61hnb1y8gc) has a much more detailed picture of what we believe that entails, but here are some highlights:
```

Comment:
Unfortunately, I don't know which requirements I'd list. That's part of why this is a hard domain - there's many, many interlinked requirements. I can see how it reads as buzzwordy, but "minimal while covering fundamental requirements" is a thing we've literally said each other during committee meetings when trying to generally size and scope this first iteration.